pkgs/development/zig-packages/build-zig-package/default.nix

```
@@ -0,0 +1,50 @@
+{ lib, stdenv, zig }:
+
+{ strictDeps ? true
+, depsBuildBuild ? [ ]
+, nativeBuildInputs ? [ ]
+, configurePhase ? null
+, buildPhase ? null
+, checkPhase ? null
+, installPhase ? null
+, enableParallelBuilding ? true
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps enableParallelBuilding;
+  depsBuildBuild = [ ] ++ depsBuildBuild;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  postConfigure = ''
+    export HOME=$TMPDIR
+  '';
+
+  buildPhase =
+    if (buildPhase == null) then ''
+      runHook preBuild
+      zig build -Drelease-safe -Dcpu=baseline --verbose
+      runHook postBuild
+    '' else
+      buildPhase;
```

and the defaults lambda paraemeter can be removed
```suggestion
  buildPhase = attrs.buildPhase or ''
    runHook preBuild
    zig build -Drelease-safe -Dcpu=baseline
    runHook postBuild
  '';
```

same suggestion for the other phases as well